### DISCUSSION TITLE
op-node l1.rpckind unknown rpc kind: "reth_db"

### DISCUSSION THREAD

**Discussion by olivain** [ethereum-optimism/developers#379] [Node Operators]
Hi there,

On my debian, I followed the instructions of this page: 
https://paradigmxyz.github.io/reth/run/optimism.html
in order to compile op-node locally with the **rethdb-reader** dylib:

```
git clone git@github.com:ethereum-optimism/optimism.git
cd optimism/op-service/rethdb-reader && cargo build --release
cd ../op-node
go build -v -tags rethdb -o ./bin/op-node ./cmd/main.go
mv bin/op-node /usr/bin/op-node

```
 Compilation went well with no issues. But when i try to run op-node with the following flags:

 --l1.rpckind=reth_db --l1.rethdb=/root/.local/share/reth_data/db

op-node fails with the following error :

>  msg="Application failed" message="invalid value \"reth_db\" for flag -l1.rpckind: unknown rpc kind: \"reth_db\""

Is the rethdb-reader still a thing ? is the tutorial i followed outdated ? Is there any way to precise to op-node that the l1 rpc runnng locally is reth ? What am i missing here ?

**Comment by hash1go** [comment]
Hello.
Could you please share the configurations you used for op-reth and op-node?

**Comment by Chomtana** [comment]
You can try with `--l1.rpckind=basic --l1.rethdb=/root/.local/share/reth_data/db` and compile with rethdb tag
